### PR TITLE
Fix format-string warning for 32-bit compilation (#1629)

### PR DESCRIPTION
**Pull request by steven-johnson**

### CODE REVIEW

File: src/binary-reader.cc [hunk 1]
```diff
@@ -629,7 +629,7 @@ Result BinaryReader::ReadAddress(Address* out_value,
                                  const char* desc) {
   ERROR_UNLESS(memory < memories.size(),
                "load/store memory %u out of range %lu", memory,
-               memories.size());
+               static_cast<unsigned long>(memories.size()));
```
Comment (sbc100) [inline src/binary-reader.cc:632]: How about changing the format string to "%zu"?

Comment (steven-johnson) [reply src/binary-reader.cc:632]: Is that 100% portable? I thought it wasn't supported in MSVC.

Comment (sbc100) [reply src/binary-reader.cc:632]: We have at least one other occurrence already so I guess its portable enough:

```
$ git grep "%zu"
src/wat-writer.cc:  WABT_TRACE_ARGS(FlushExprTreeVector, "%zu", expr_trees.size());
```

If it proves non-portable then the correct fix is probably to add a PRI macro like we do for `PRIaddress`

Comment (sbc100) [reply src/binary-reader.cc:632]: According to this discussion MS added support for %z in VS2013: https://stackoverflow.com/questions/15610053/correct-printf-format-specifier-for-size-t-zu-or-iu

Comment (steven-johnson) [reply src/binary-reader.cc]: Done.

**Review by sbc100 — APPROVED** [review]

**Pull request merged**